core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -163,6 +163,17 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
     info(s"Added fetcher to broker ${fetcherThread.sourceBroker.id} for partitions $initialOffsetAndEpochs")
   }
 
+  def addTopicIdsToFetcherThread(partitionsToUpdate: Set[(TopicPartition, BrokerAndFetcherId)], topicIds: String => Option[Uuid]): Unit = {
+    lock synchronized {
+      val partitionsPerFetcher = partitionsToUpdate.groupMap(_._2)(_._1)
+
+      for ((brokerAndFetcherId, partitions) <- partitionsPerFetcher) {
+        val brokerIdAndFetcherId = BrokerIdAndFetcherId(brokerAndFetcherId.broker.id, brokerAndFetcherId.fetcherId)
```

Comment:
It is a bit weird that you have to recreate `BrokerIdAndFetcherId` from `BrokerIdAndFetcherId` here.

Reply:
I was pulling from the add partitions method. But it does something different. 🤦‍♀️
I can just use the one from before.

Reply:
Ah actually this is really subtle. One is BrokerAndFetcherId and the other is Broker**_Id_**AndFetcherId. I can simplify and just create the second one.

Reply:
Yeah, that's right. I think that an important point is to let the FetcherManager compute the fetcher id based on the topic-partition instead of computing it in the replica manager.

Reply:
Makes sense. This will be in the next commit.